tests/hardhat/XVS/XVSVaultFix.ts

```diff
@@ -0,0 +1,218 @@
+  await xvsVaultImpl.deployed();
+
+  await xvsVaultProxy.connect(admin)._setPendingImplementation(xvsVaultImpl.address);
+  await xvsVaultImpl.connect(admin)._become(xvsVaultProxy.address);
+  xvsVault = XVSVault__factory.connect(xvsVaultProxy.address, admin);
+}
+
+async function deployAndConfigureOldVault() {
+  /*
+   *  Forking mainnet
+   * */
+  await impersonateAccount(Owner);
+  admin = await ethers.getSigner(Owner);
+
+  const xvsVaultProxy = XVSVaultProxy__factory.connect(vaultProxy, admin);
+  oldXVSVault = XVSVault__factory.connect(xvsVaultProxy.address, admin);
+}
+
+async function sendGasCost() {
+  /**
+   *  sending gas cost to owner
+   * */
+  [signer] = await ethers.getSigners();
+  console.log("-- Sending gas cost to owner addr --");
+  await signer.sendTransaction({
+    to: Owner,
+    value: ethers.BigNumber.from("900081987000000000"),
+    data: undefined,
+  });
+}
+
+describe.only("XVSVault", async () => {
+  before(async () => {
+    getForkingUrl();
+  });
+  if (FORK_MAINNET) {
+    it("Verify Expected Result old withdraw requests for affectedUser1", async () => {
+      await reset(`${FORK_ENDPOINT}`, 25458046);
+      await sendGasCost();
+      await deployAndConfigureNewVault();
+
+      const result = await xvsVault.getWithdrawalRequests(tokenAddress, poolId, affectedUserAddress);
+
+      expect(result[0][0].toString()).to.eql(beforeUpgrade[0][0].toString());
+      expect(result[0][1].toString()).to.eql(beforeUpgrade[0][1].toString());
+      expect(result[0][2].toString()).to.eql(beforeUpgrade[0][2].toString());
+
+      expect(result[1][0].toString()).to.eql(beforeUpgrade[1][0].toString());
+      expect(result[1][1].toString()).to.eql(beforeUpgrade[1][1].toString());
+      expect(result[1][2].toString()).to.eql(beforeUpgrade[1][2].toString());
+
+      expect(result[2][0].toString()).to.eql(beforeUpgrade[2][0].toString());
+      expect(result[2][1].toString()).to.eql(beforeUpgrade[2][1].toString());
+      expect(result[2][2].toString()).to.eql(beforeUpgrade[2][2].toString());
+
+      expect(result[3][0].toString()).to.eql(beforeUpgrade[3][0].toString());
+      expect(result[3][1].toString()).to.eql(beforeUpgrade[3][1].toString());
+      expect(result[3][2].toString()).to.eql(beforeUpgrade[3][2].toString());
+
+      expect(result[4][0].toString()).to.eql(beforeUpgrade[4][0].toString());
+      expect(result[4][1].toString()).to.eql(beforeUpgrade[4][1].toString());
+      expect(result[4][2].toString()).to.eql(beforeUpgrade[4][2].toString());
+    });
+
+    it("Execute withdrawal", async () => {
+      await reset(`${FORK_ENDPOINT}`, 25458046);
+      await sendGasCost();
+      await deployAndConfigureNewVault();
+      await getForkingUrl();
+
+      await impersonateAccount(affectedUserAddress);
+      const affectedUser1Signer = await ethers.getSigner(affectedUserAddress);
+
+      await xvsVault.connect(affectedUser1Signer).executeWithdrawal(tokenAddress, poolId);
```

Comment:
We might check another side effect of this action, like the emission of the `ExecutedWithdrawal` event with the expected amount